### PR TITLE
chore: change import path to v4

### PR DESCRIPTION
**Pull request by johnletey**
This PR changes our import path to include `v4` to correctly align ourselves with Go's [naming convention](https://go.dev/doc/modules/version-numbers).

It requires the dependency work done in #259, and should be merged after `v4.0.0` is released!

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -152,6 +151,9 @@ require (
 )
 
 replace (
+	github.com/circlefin/noble-cctp => github.com/noble-assets/cctp v0.0.0-20231017104045-821e63c6c4cb
+	github.com/circlefin/noble-fiattokenfactory => github.com/noble-assets/fiattokenfactory v0.0.0-20231017103812-a2afc92603fc
```
Comment (agouin) [inline go.mod:155]: I assume we will remove these replaces and the associated changes like the GITAUTH -> SSH_NOBLE before merging?

File: proto/globalfee/query.proto [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-import "cosmos/base/v1beta1/coin.proto";
-import "globalfee/genesis.proto";
 
-option go_package = "github.com/strangelove-ventures/noble/x/globalfee/types";
+option go_package = "github.com/strangelove-ventures/noble/v4/x/globalfee/types";
 
 // Query defines the gRPC querier service.
 service Query {
-  rpc Params(QueryParamsRequest) returns (Params) {
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
```
Comment (boojamya) [inline proto/globalfee/query.proto:12]: 👍

**Review by boojamya — APPROVED** [review]
Looks good!

**Comment by johnletey** [comment]
@Mergifyio backport release/v4.1.x

**Pull request merged**